### ISSUE TITLE
AssemblyVersionInformation breaks fsc --standalone

### ISSUE THREAD

**Issue by 0x53A** [fsharp/FAKE#1637]
I was trying to compile paket with --standalone

![image](https://user-images.githubusercontent.com/4236651/29499867-47bfb604-861a-11e7-9767-63bf22cd09d8.png)

As I understand it, the issue is that Fake generates the class in the System namespace:

![image](https://user-images.githubusercontent.com/4236651/29499876-6799bb0a-861a-11e7-8da6-9266c6c03506.png)

That means that if I have multiple projects using this, i have a duplicated type.

In the case of Paket it is probably Argu that it conflicts with.

Fix: use a custom (configurable) namespace.

**Comment by matthid** [comment]
I think there is a way to unblock yourself by generating a different namespace. It depends on what API you use for AssemblyInfo generation. The default is System, and any default has the same problem...

**Comment by 0x53A** [comment]
So I didn't look at the actual Fake source ....


This is the Method that Paket uses:

![image](https://user-images.githubusercontent.com/4236651/29519620-6031c3da-867e-11e7-98a0-cca1d44dc655.png)


Here you can pass a filename and a bunch of Attributes, but I don't see a way to pass a namespace (and there is no Attribute "namespace", which wouldn't make sense anyway).

---------

Also, the bigger issue is that dependencies with this poison an unrelated project. by that I mean if I use two projects which both use Fake ``CreateFSharpAssemblyInfo``, but I myself don't use it, then it still conflicts.

So I propose to add new APIs which DO take a namespace, deprecate the existing ones, and then send pull-request to most major open source projects ...

**Comment by matthid** [comment]
Afaik this was not an issue, because by default the types are internal? Or did we change this recently?

**Comment by matthid** [comment]
Here is the config: https://github.com/fsharp/FAKE/blob/master/src/app/Fake.DotNet.AssemblyInfoFile/AssemblyInfoFile.fs#L53
And here the member to use: https://github.com/fsharp/FAKE/blob/master/src/app/Fake.DotNet.AssemblyInfoFile/AssemblyInfoFile.fs#L242

**Comment by 0x53A** [comment]
> Afaik this was not an issue, because by default the types are internal? Or did we change this recently?

Yes, they are internal, but if you do ``fsc --standalone``, then they still need to get merged into one assembly.

paket uses ILRepack, which somehow handles this, but I'm not sure how. Either it prefixes all types with the original assembly name, or it just drops duplicates, or ...

Thanks for looking up the correct method to use when I was too lazy ;)

**Comment by matthid** [comment]
I'm assuming you could unblock youself with the available options, let me know if we still want to do anything.
